src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/diagnostics/rules/IllegalPackageAPIExportsDiagnosticRule.java

```diff
@@ -51,18 +44,20 @@ public void scanIndividual(final CompilationUnit cu, final APIListing listing) {
                         validateType(methodId, returnType, listing);
                     }
 
-                    methodDecl.getParameters().stream()
-                            .map(Parameter::getType)
-                            .filter(Type::isClassOrInterfaceType)
-                            .map(Type::asClassOrInterfaceType)
-                            .forEach(parameter -> validateType(methodId, parameter, listing));
+                    validateParameters(methodDecl, methodId, listing);
                 });
     }
 
     private void validateType(String methodName, ClassOrInterfaceType type, final APIListing listing) {
         String typeAsString = type.getNameAsString();
 
-        if (listing.getTypeToPackageNameMap().containsKey(typeAsString)) {
+        if (typeAsString.contains(".")) {
```

Comment:
This may have false positives if the type is a nested static class like `ReactorNettyClient.ReactorNettyHttpResponse` which will try to validate the outer class as package (ReactorNettyClient).